src/Block/Types.hs

```diff
@@ -62,23 +82,32 @@ instance (ToJSON args, ToJSON mirror) => ToJSON (OgmiosRequest args mirror) wher
   toJSON =
     Aeson.genericToJSON Aeson.defaultOptions {Aeson.fieldLabelModifier = drop 1}
 
-type OgmiosFindIntersectRequest = OgmiosRequest CursorPoints OgmiosMirror
+type OgmiosFindIntersectRequest = OgmiosRequest [Text] OgmiosMirror
 
 type OgmiosRequestNextRequest = OgmiosRequest (Map Text Text) OgmiosMirror
 
 mkFindIntersectRequest :: BlockInfo -> OgmiosFindIntersectRequest
```

Comment:
If `ToJSON` instance is changed, this function wouldn't need case split and will work on both constructors out of the box

Reply:
I've simplified the case split but I don't think you can get rid of it entirely.